tests/service/data-util.service.spec.ts

```diff
@@ -16,55 +16,92 @@
-            const tempLink = document.createElement('a');
-            jest.spyOn(tempLink, 'click');
-            jest.spyOn(document, 'createElement').mockReturnValue(tempLink);
-            // call downloadFile function
-            // csv content:
-            // ID,Name
-            // 1,Toto
-            const contentType = 'text/csv';
-            const data = 'SUQsTmFtZQ0KMSxUb3Rv';
-            const fileName = 'test-download-file.csv';
-            service.downloadFile(contentType, data, fileName);
-            expect(document.createElement).toHaveBeenCalledTimes(1);
-            expect(document.createElement).toHaveBeenCalledWith('a');
-            expect(tempLink.target).toBe('_blank');
-            expect(tempLink.download).toBe('test-download-file.csv');
-            expect(tempLink.click).toHaveBeenCalledTimes(1);
-            expect(tempLink.click).toHaveBeenCalledWith();
-        }));
     });
+
+    it('should not abbreviate the text when below cutoff', inject([JhiDataUtils], (service: JhiDataUtils) => {
+        expect(service.abbreviate('Hello Jhipster')).toBe('Hello Jhipster');
+    }));
+
+    it('should abbreviate the text and append ...', inject([JhiDataUtils], (service: JhiDataUtils) => {
+        expect(service.abbreviate('Hello Jhipster lets test the data utils function')).toBe('Hello Jhipster ...s function');
+    }));
+
+    it('should abbreviate the text and append +++', inject([JhiDataUtils], (service: JhiDataUtils) => {
+        expect(service.abbreviate('Hello Jhipster lets test the data utils function', '+++')).toBe('Hello Jhipster +++s function');
+    }));
+
+    it('should return the bytesize of the text', inject([JhiDataUtils], (service: JhiDataUtils) => {
+        expect(service.byteSize('Hello Jhipster')).toBe(`10.5 bytes`);
+    }));
+
+    it('should download the csv file', inject([JhiDataUtils], (service: JhiDataUtils) => {
+        const tempLink = document.createElement('a');
+        jest.spyOn(tempLink, 'click');
+        jest.spyOn(document, 'createElement').mockReturnValue(tempLink);
+        // call downloadFile function
+        // csv content:
+        // ID,Name
+        // 1,Toto
+        const contentType = 'text/csv';
+        const data = 'SUQsTmFtZQ0KMSxUb3Rv';
+        const fileName = 'test-download-file.csv';
+        service.downloadFile(contentType, data, fileName);
+        expect(document.createElement).toHaveBeenCalledTimes(1);
+        expect(document.createElement).toHaveBeenCalledWith('a');
+        expect(tempLink.target).toBe('_blank');
+        expect(tempLink.download).toBe('test-download-file.csv');
+        expect(tempLink.click).toHaveBeenCalledTimes(1);
+        expect(tempLink.click).toHaveBeenCalledWith();
+    }));
+
+    it('should return a promise that rejects with an error message when image is passed but file type is not image', async(() => inject([JhiDataUtils], (service: JhiDataUtils) => {
```

Comment:
please change `async` function usage to below syntax:

> it('should return a promise that rejects with an error message when image is passed but file type is not image', `async(inject([JhiDataUtils], (service: JhiDataUtils) => {`

Reply:
done